zebra/zebra_pw.c

```diff
@@ -142,18 +142,6 @@ check_lsp (struct zebra_pw_t *pw)
       zlog_warn ("No rib found for PW %u at VPN %s", pw->pwid, pw->vpn_name);
       return 1;
     }
-  /* check labels for each nexthop in Route */
-  /*
-   * Need to ensure that there's a label binding for all nexthops.
-   * Otherwise, ECMP for this route could render the pseudowire unusable.
-   */
-  for (ALL_NEXTHOPS_RO(rib->nexthop, nexthop, tnexthop, recursing))
-    if (!nexthop->nh_label || nexthop->nh_label->num_labels == 0)
-      {
-        zlog_warn ("No label found in rib for PW %u at VPN %s", pw->pwid,
-                   pw->vpn_name);
-        return 1;
-      }
```

Comment:
It should be safe.

Back when I wrote this code I thought we could avoid making unnecessary changes in the RIB because implicit-null labels are never sent on the wire. But it turns out that we need to keep track of implicit-null labels in the RIB because of this issue you brought up. `rt_netlink.c` and `zebra_mpls_openbsd.c` have additional checks that will ensure that the implicit-null labels are never installed in the kernel.